src/blueapi/core/context.py

```diff
@@ -63,9 +64,10 @@ def find_device(self, addr: Union[str, List[str]]) -> Optional[Device]:
         else:
             return find_component(self.devices, addr)
 
-    def with_startup_script(self, path: Union[Path, str]) -> None:
-        mod = import_module(str(path))
-        self.with_module(mod)
+    def with_config(self, config: EnvironmentConfig) -> None:
+        for source in config.sources:
+            mod = import_module(str(source.module))
+            self.with_module(mod)
```

Comment:
We want to be doing a different action based on the value of `source.type`, so something like:
```python
if source.type == "plans":
    self.with_plan_module(mod)
elif source.type == "dodal":
    self.with_device_module(mod)
else:
    ...
```